aspnet/fundamentals/middleware/sample/src/MiddlewareSample/RequestLoggerExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Hosting;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+
+namespace MiddlewareSample
+{
+    public static class RequestLoggerExtensions
+    {
+        public static IApplicationBuilder UseRequestLogger(this IApplicationBuilder builder)
+        {
+            var env = builder.ApplicationServices.GetService<IHostingEnvironment>();
```

Comment:
See https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication/AuthenticationMiddleware.cs#L35 for our recommended way to set up logging. I wouldn't want anyone to copy this sample code.